### PR TITLE
Slider: Implement Classes Option

### PR DESCRIPTION
**Pull request by cgack**
Fixes gh-7716

### CODE REVIEW

File: js/widgets/forms/slider.js [hunk 1]
```diff
@@ -409,40 +322,31 @@ return $.widget( "mobile.slider", $.extend( {
...
 			left, width, data, tol,
 			pxStep, percent,
-			control, isInput, optionElements, min, max, step,
+			control, min, max, step,
 			newval, valModStep, alignValue, percentPerStep,
 			handlePercent, aPercent, bPercent,
 			valueChanged;
 
-		self.slider[ 0 ].className = [ this.isToggleSwitch ? "ui-slider ui-slider-switch ui-slider-track ui-shadow-inset" : "ui-slider-track ui-shadow-inset", trackThemeClass, cornerClass, miniClass ].join( "" );
+		this._addClass( $( self.slider[ 0 ] ), "ui-slider-track", trackThemeClass );
```
Comment (arschmitz) [inline js/widgets/forms/slider.js]: cant this just be self.slider seems like your accessing the element then re creating a new jQuery object for it?

File: js/widgets/forms/slider.js [hunk 2]
```diff
@@ -409,40 +322,31 @@ return $.widget( "mobile.slider", $.extend( {
...
 		this.value = this._value();
-		if ( this.options.highlight && !this.isToggleSwitch && this.slider.find( ".ui-slider-bg" ).length === 0 ) {
-			this.valuebg = ( function() {
-				var bg = document.createElement( "div" );
-				bg.className = "ui-slider-bg " + $.mobile.activeBtnClass;
-				return $( bg ).prependTo( self.slider );
-			} )();
-		}
-		this.handle.addClass( "ui-button" + themeClass + " ui-shadow" );
+		this._addClass( this.handle, null, "ui-button" + themeClass + " ui-shadow" );
```
Comment (arschmitz) [inline js/widgets/forms/slider.js]: Shouldnt there be a same after `ui-button`?

File: tests/unit/slider/index.html [hunk 3]
```diff
@@ -13,6 +13,8 @@
 	<script src="../../../external/qunit-assert-classes/qunit-assert-classes.js"></script>
 	<script src="../../../tests/jquery.testHelper.js"></script>
 	<script>
+		// $.mobileBackcompat = false;
```
Comment (arschmitz) [inline tests/unit/slider/index.html]: shouldn't this be uncommented?

File: tests/unit/slider/slider_backcompat.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	var $highlighted = $( "#background-slider" ),
+		$unhighlighted = $( "#no-background-slider" );
+
+	deepEqual( $highlighted.siblings( ".ui-slider-track" ).find( ".ui-slider-bg" ).length, 1,
+		"highlighted slider should have a div for the track bg" );
+	deepEqual( $unhighlighted.siblings( ".ui-slider-track" ).find( ".ui-slider-bg" ).length, 0,
+		"unhighlighted slider _not_ should have a div for the track bg" );
+} );
+
+} )( jQuery );
```
Comment (arschmitz) [inline tests/unit/slider/slider_backcompat.js]: missing new line. Also shouldn't there be tests for corner and mini here?

**Comment by arschmitz** [comment]
Your never adding the classes to the textinput to style it.

File: js/widgets/forms/slider.js [hunk 5]
```diff
@@ -43,52 +43,37 @@ return $.widget( "mobile.slider", $.extend( {
...
-			} )() : false,
-			options,
-			wrapper,
-			j, length,
-			i, optionsCount, origTabIndex,
-			side, activeClass, sliderImg;
+			wrapper;
 
 		$label.attr( "id", labelID );
-		this.isToggleSwitch = isToggleSwitch;
```
Comment (arschmitz) [inline js/widgets/forms/slider.js]: this is not new is actually a bug but... In create we should be calling
```js
if ( $.mobile.textinput ) {
    this.element.textinput();
}
```

**Comment by arschmitz** [comment]
Your adding the theme extension here but never calling or using it.

**Comment by arschmitz** [comment]
@cgack ready for some updates

**Comment by cgack** [comment]
@arschmitz I've made some adjustments. ready for another round of review.

File: js/widgets/forms/slider.backcompat.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+			}
+		},
+
+		_setCorners: function( value ) {
+			this._toggleClass( this.slider, null, "ui-corner-all", value );
+			this._toggleClass( this.element, null, "ui-corner-all", value );
+		}
+	} );
+
+	$.widget( "mobile.slider", $.mobile.slider, $.mobile.widget.backcompat );
```
Comment (arschmitz) [inline js/widgets/forms/slider.backcompat.js]: corners, mini and highlight are all boolean options and should be handled by backcompat

File: js/widgets/forms/slider.js (hunk 5 above)
Comment (arschmitz) [inline js/widgets/forms/slider.js]: You still have not addressed this

**Comment by arschmitz** [comment]
@cgack it does not look like you addressed the issues with the theme extension or with the styling of the textinputs.

**Comment by cgack** [comment]
@arschmitz I rebased the theme updates and made the theme changes. time for another review

**Comment by arschmitz** [comment]
@cgack it looks like your rebase is a little off it includes some of my commits

**Comment by cgack** [comment]
@arschmitz cherry-pick complete, should be cleaner to review this time :)

**Comment by arschmitz** [comment]
@cgack thank you reviewing right now

File: js/widgets/forms/slider.backcompat.js (hunk 6 above)
Comment (arschmitz) [inline js/widgets/forms/slider.backcompat.js]: This is still not addressed

File: js/widgets/forms/slider.js (hunk 5 above)
Comment (arschmitz) [inline js/widgets/forms/slider.js]: Bump!

File: js/widgets/forms/slider.backcompat.js (hunk 6 above)
Comment (cgack) [inline js/widgets/forms/slider.backcompat.js]: @arschmitz not sure what you mean here. This code is required to get the backcompat to function properly because the way the slider is styled it doesn't automatically get the required classes for these properties.

**Comment by arschmitz** [comment]
This looks good now i just need to do some manual testing and i think this is good to land

**Comment by arschmitz** [comment]
:+1:

**Comment by cgack** [comment]
Closed via 36becfe52c619d71282c41aa28995bd6d1a327b8 on 1.5-dev
